Sources/NestCLI/Executor/RunCommandExecutor.swift

```diff
@@ -0,0 +1,64 @@
+import NestKit
+import Logging
+
+public struct RunCommandExecutor {
+    /// `owner/repo` format
+    public let reference: String
+    public let subcommands: [String]
+    
+    public enum ParseError: Error {
+        case emptyArguments
+        case invalidFormat
+    }
+    
+    public init(arguments: [String]) throws {
+        guard !arguments.isEmpty else {
+            throw ParseError.emptyArguments
+        }
+        guard arguments[0].contains("/") else {
+            throw ParseError.invalidFormat
+        }
+        
+        self.reference = arguments[0]
+        self.subcommands = if arguments.count >= 2 {
+            Array(arguments[1...])
+        } else {
+            []
+        }
+    }
+    
+    /// Returns a relative path. If not, it will attempt to install.
+    public func resolveBinaryRelativePath(
+        noInstall: Bool,
+        reference: String,
+        version: String,
+        target: Nestfile.Target,
+        gitURL: GitURL,
+        gitVersion: GitVersion,
+        nestInfoController: NestInfoController,
+        executableBinaryPreparer: ExecutableBinaryPreparer,
+        artifactBundleManager: ArtifactBundleManager,
+        logger: Logger
+    ) async throws -> String? {
```

Comment:
Ideally, I would write tests, but there are no tests for the dependent component (`ExecutableBinaryPreparer`), so I decided not to write tests for the `resolveBinaryRelativePath` method.

(If it is required, I will write tests together with `ExecutableBinaryPreparer`!)